### DISCUSSION TITLE
Question on /v1/graph route

### DISCUSSION THREAD

**Discussion by Funasitien** [plan-player-analytics/Plan#3331] [APIs and Development]
I was playing a bit with API and see that a graph route exist. So wanted to ask first which graph types are allowed (even with the link specified in the Sawager doc, some of the graph types in the file seems to not work, or I use them wrongly idk). And then I wanted to know if it's possible to convert the raw data to an actual image in python, and if you have any packages to suggest to me. Thanks!

**Comment by AuroraLS3** [comment]
Here's all the types https://github.com/plan-player-analytics/Plan/blob/master/Plan/common/src/main/java/com/djrapitops/plan/delivery/webserver/resolver/json/GraphsJSONResolver.java#L172

Note that same endpoint is used for both server and network so you may need server=uuid as query parameter in your URL where network version of a graph doesn't exist.

You can use plotly in python for generating graph images.

**Reply by Funasitien** [reply]
Thanks!
